.github/workflows/build-plugin-archive.yml

```diff
@@ -118,38 +104,82 @@ jobs:
               .parse(process.env.ENV_VARS)
               .forEach(envVar => core.exportVariable(envVar.name, envVar.value));
 
-      - name: Set up node cache mode
-        run: |
-          if [ "${{ inputs.PACKAGE_MANAGER }}" == 'npm' ] && { [ -f "${GITHUB_WORKSPACE}/package-lock.json" ] || [ -f "${GITHUB_WORKSPACE}/npm-shrinkwrap.json" ]; }; then
-            echo "NODE_CACHE_MODE=npm" >> $GITHUB_ENV
-          elif [ "${{ inputs.PACKAGE_MANAGER }}" == 'yarn' ] && [ -f "${GITHUB_WORKSPACE}/yarn.lock" ]; then
-            echo "NODE_CACHE_MODE=yarn" >> $GITHUB_ENV
-          else
-            echo "No lock files found or unknown package manager"
-          fi
-
       - name: Set up PHP
         uses: shivammathur/setup-php@v2
         with:
           php-version: ${{ inputs.PHP_VERSION }}
           tools: wp-cli
 
-      - name: Install Composer dependencies
-        # ⬑ The idea is to ensure we have non-production tools like Composer Asset Compiler and WordPress Translation Downloader available
+      - name: Install Composer dependencies without dev dependencies
         uses: ramsey/composer-install@v3
         with:
-          composer-options: --prefer-dist
+          composer-options: ${{ inputs.COMPOSER_ARGS }}
+
+      - name: Set artifact name
+        id: set-artifact-name
+        run: echo "artifact=interim-deps" >> $GITHUB_OUTPUT
 
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ steps.set-artifact-name.outputs.artifact }}
+          path: |
+            ./*
+            !./.git
+            !./.ddev
+            !./.github
+  run-build-tools:
+    name: Process build steps
+    timeout-minutes: 5
+    runs-on: ubuntu-latest
+    needs: checkout-dependencies
+    env:
+      NODE_OPTIONS: ${{ inputs.NODE_OPTIONS }}
+      NODE_AUTH_TOKEN: ${{ secrets.NPM_REGISTRY_TOKEN }}
+      NODE_CACHE_MODE: ''
+    outputs:
+      artifact: ${{ steps.set-artifact-name.outputs.artifact }}
+    steps:
+      - name: Download Artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ needs.checkout-dependencies.outputs.artifact }}
       - name: Check optional Composer build tools
         id: composer-tools
         run: |
+          hasAssetConfig(){
+            test -f assets-compiler.json
+            local EXIT=$?
+            if [ ! $EXIT ]; then
+              echo "$EXIT"
+              exit 0
+            fi
+            jq '.extra | has("composer-asset-compiler")' < composer.json >/dev/null 2>&1
+            local EXIT=$?
+            echo "$EXIT"
+          }
+          hasTranslateConfig(){
+            jq '.extra | has("wp-translation-downloader")' < composer.json >/dev/null 2>&1
+            local EXIT=$?
+            echo "$EXIT"
+          }
           hasDep(){
             composer show -i -N -D --strict "${1}" >/dev/null 2>&1
             local EXIT=$?
             echo "$EXIT"
           }
```

Comment:
`hasDep()` is no longer used.
```suggestion
```

Reply:
Yes but I am letting go of this awesome piece of scripting with a heavy heart, just so you know